packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -641,6 +663,28 @@ function propagateParentContextChanges(
   workInProgress.flags |= DidPropagateContext;
 }
 
+function checkIfComparedContextValuesChanged(
+  oldComparedValue: mixed,
+  newComparedValue: mixed,
+): boolean {
+  if (isArray(oldComparedValue) && isArray(newComparedValue)) {
```

Comment:
Let's comment why this array check is safe. You mentioned in the comments but would be good to clarify that returning an array an implicit contract and we don't expect to return other things for now

Reply:
Updated